pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
> 1. Using the CLI, how will an end user set "the command to run tests"?

If we expose it that way at all, i'd expect something like: 
`$ oc new-build <blah blah> --post-commit-cmd="rake test"`

> 2. Same question as above but for the Web Console.

the UI team has been addding editable fields for various BuildConfig settings, i'd expect another field for this.

> I had in mind a single text area just like other CIs offer. If we have a way to setup the entrypoint and cmd, and those are string slices, it's harder for me to imagine a good UI/UX.

a string is fine, we can spliton spaces to construct the slice of args.  As for having two strings (entrypoint and cmd), i don't see a good alternative other than doing what was discussed above:  If the first arg of the "cmd" starts with "/" then treat it as the entrypoint, otherwise use "bash -c" as the entrypoint.